tox.ini

```
@@ -62,63 +68,29 @@ commands_post =
 max-line-length = 99
 exclude = .tox,wiki,.cache,.d3a,.hypothesis,.pytest_cache,vagrant,requirements,venv
 
-[testenv:codecoverage]
-basepython = python3.8
-passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH SOLC_BINARY
-deps =
-    pip-tools
-    coverage
-commands =
-    pip-sync requirements/tests.txt
-    pip install -e .
-    coverage run -m pytest --random-order {posargs:tests}
-    coverage combine
-    coverage report
 
 [testenv:ci]
-basepython = python3.8
-passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH SOLC_BINARY GSY_FRAMEWORK_BRANCH INTEGRATION_TESTS_REPO INTEGRATION_TESTS_BRANCH
-deps =
-    pip-tools
-    coverage
-allowlist_externals =
-    git
-    /bin/rm
-    /bin/ln
-    bash
 commands_pre =
     python -c "import fcntl; fcntl.fcntl(1, fcntl.F_SETFL, 0)"
-    pip-sync requirements/tests.txt requirements/pandapower.txt
-    pip install -e .
+    pip-sync requirements/pandapower.txt
+    {[testenv:integrationtests]commands_pre}
     pip uninstall -y gsy-framework
     pip install git+https://github.com/gridsingularity/gsy-framework@{env:GSY_FRAMEWORK_BRANCH:master}
 commands =
     flake8
```

I know it's the same, but this would show that we're re-using another env.
```suggestion
    {[testenv:flake8]commands}
```
(To be honest, I'm not completely sure that we need the `testenv:flake8` environment at all in our tox file).

---

Ok. I'll adapt it. But if it is not necessary at all I could better remove it. 
Could it be maybe needed for the pre-commit workflow?

---

Nope, that should work independently as far as I know. I think we can remove the `[testenv:flake8]` env.

The only part we should keep is the `[flake8]` one: https://github.com/gridsingularity/gsy-e/blob/228599c33ba283021d1357309bec3f71428da739/tox.ini#L61-L63

because that's actually the flake8 configuration we're using. I'm not particularly fond of having it inside the `tox.ini` file, but that's ok for the moment.

---

Ok, let's do it that way then. I agree with you.